### PR TITLE
General ui layout revised

### PR DESCRIPTION
**Pull request by JustinPraas**
Revised the layout using flex boxes. 

An example on how to use the layout can be found in Assignments.tsx.

### CODE REVIEW

File: frontend/src/components/NavigationBarOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+}
+
+interface NavigationBarOverlayState { }
+
+class NavigationBarOverlay extends Component<NavigationBarOverlayProps, NavigationBarOverlayState> {
+
+    componentDidMount() {
+        // Fetch the assignment sets data
+        this.props.fetchAssignmentSetDtoBriefs(20); // TODO: unhardcode course ID
```
Comment (harindup) [inline frontend/src/components/NavigationBarOverlay.tsx]: Why is this hardcoded? I think this should be taken from the URL, for example: /courses/:courseId/assignments

File: frontend/src/state/courses/assignments/sagas.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { put, takeEvery, call } from "redux-saga/effects";
+import { ASSIGNMENT_SET_DTO_BRIEFS_FETCH_REQUESTED_ACTION } from "./constants";
+import { AssignmentSetDtoFetchAction, assignmentSetDtoBriefsFetchSucceededAction } from "./actions";
+
+import { authenticatedFetchJSON } from "../../../api/sagas";
+import { AssignmentSetDtoBrief } from "../../types";
+
+export function* fetchAssignmentSetDtoBriefs(action: AssignmentSetDtoFetchAction) {
```
Comment (harindup) [inline frontend/src/state/courses/assignments/sagas.ts]: What happens during a failure? Remco has handled this in the courses.

File: frontend/src/state/courses/assignments/constants.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export const ASSIGNMENT_SET_DTO_BRIEFS_FETCH_REQUESTED_ACTION =
```
Comment (harindup) [inline frontend/src/state/courses/assignments/constants.ts]: Failure states?

File: frontend/src/state/courses/assignments/types.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { AssignmentSetDtoBrief } from "../../types";
+
+export interface AssignmentSetsState {
```
Comment (harindup) [inline frontend/src/state/courses/assignments/types.ts]: Error state?

File: frontend/src/components/pages/assignments/AssignmentSetListEntry.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            <Badge className="badge-primary" key={groupSet.id}>{groupSet.name}</Badge>);
+
+        return (
+            <Card className="my-3">
+                <CardHeader>
+                    {this.props.assignmentSetTitle}
+                    <Button style={{'float':'right'}} color="primary" size="sm"><FontAwesomeIcon icon={faEdit}/> Edit</Button>
+                </CardHeader>
+                <CardBody>
+                    <p>Assigned group sets: { linkedGroupSetTags }</p>
```
Comment (harindup) [inline frontend/src/components/pages/assignments/AssignmentSetListEntry.tsx]: When the groups are empty, this should show (in grayed italics) "No groups assigned. Click on Edit to assign groups."

File: frontend/src/components/NavigationBarOverlay.tsx (hunk 1 above)
Comment (JustinPraas) [reply frontend/src/components/NavigationBarOverlay.tsx]: You're right. Forgot!

File: frontend/src/state/courses/assignments/sagas.ts (hunk 2 above)
Comment (JustinPraas) [reply frontend/src/state/courses/assignments/sagas.ts]: Will dive into this soon.

**Comment by VriesDeRick** [comment]
Does this fix either #20 or @22?

File: frontend/src/components/NavigationBarOverlay.tsx (hunk 1 above)
Comment (JustinPraas) [reply frontend/src/components/NavigationBarOverlay.tsx]: After hours of looking why nested URLs were not allowed, I got help on stackoverflow and managed to fix the issue by adjusting the webpack.config.js. I added _publicPath: '/',_ to the _output_ object in _module.exports_.

**Comment by JustinPraas** [comment]
@VriesDeRick this will not fix #22, neither does this PR have anything to do with #22. 

However, this PR does not contain the full functionality that will fix #20, since it will also require the functionality to add and edit the assignment sets.

**Pull request merged**